Enable/Disable the Spheres in the scene (Benchmark app)

elviscapiaq opened the pull request:
- It also changes the visibility of knobs related to the spheres accordingly.

angela28chen commented on benchmarks/graphics_pipeline/GraphicsBenchmarkApp.cpp:
```diff
@@ -659,8 +679,12 @@ void GraphicsBenchmarkApp::RecordCommandBuffer(PerFrame& frame, grfx::SwapchainP
 
     // Record commands for the scene using one renderpass
     frame.cmd->BeginRenderPass(currentRenderPass);
```
If both the skybox and the sphere are disabled, I think this renderpass should not be begin/ended?

angela28chen commented on benchmarks/graphics_pipeline/GraphicsBenchmarkApp.cpp:
```diff
@@ -53,6 +53,22 @@ void GraphicsBenchmarkApp::InitKnobs()
     pDrawCallCount->SetDisplayName("DrawCall Count");
     pDrawCallCount->SetFlagDescription("Select the number of draw calls to be used to draw the `sphere-count` spheres.");
 
+    pAlphaBlend = GetKnobManager().CreateKnob<ppx::KnobCheckbox>("alpha-blend", false);
```
Can you also change the `Indent` of all the knobs related to the spheres to 1? Similar to how the quads related knobs are indented and displayed below the `pFullscreenQuadsCount` knob.

elviscapiaq replied:
Done.

wangra-google commented on benchmarks/graphics_pipeline/GraphicsBenchmarkApp.cpp:
```diff
@@ -659,8 +679,12 @@ void GraphicsBenchmarkApp::RecordCommandBuffer(PerFrame& frame, grfx::SwapchainP
     frame.cmd->BeginRenderPass(currentRenderPass);
-    RecordCommandBufferSkybox(frame);
-    RecordCommandBufferSpheres(frame);
+    if (pEnableSkyBox->GetValue()) {
```
we need to disable the `currentRenderPass` when both skybox and spheres are disabled?

elviscapiaq replied:
Done.

wangra-google replied:
sorry i couldn't find this change to disable the render pass when both both skybox and spheres are disabled, is it in a different PR?

elviscapiaq replied:
Sorry, when I rebased this PR with the last commits, I realized the Enable/Disable knobs for Skybox and Spheres were not working as it used to be and it seems the issue is related to recent commits. But now they were fixed with https://github.com/google/bigwheels/pull/331, so I will wait to this to be merged to update.

elviscapiaq replied:
Updated!

elviscapiaq commented:
I fixed the issues with the render passes: when nothing is rendered, then the renderpass for GUI should be cleared (ATTACHMENT_LOAD_OP_CLEAR) and also I rebased it.

elviscapiaq commented:
I rebased it with the main branch.

angela28chen commented on benchmarks/graphics_pipeline/GraphicsBenchmarkApp.cpp:
```diff
@@ -25,13 +25,23 @@ void GraphicsBenchmarkApp::InitKnobs()
     pKnobPs->SetFlagDescription("Select the pixel shader for the graphics pipeline.");
+    pKnobPs->SetIndent(1);
 
     pAllTexturesTo1x1 = GetKnobManager().CreateKnob<ppx::KnobCheckbox>("all-textures-to-1x1", false);
```
Should be indent of 2 now

angela28chen reviewed (approved):
LGTM, thanks for incorporating the changes for new knob pAllTexturesTo1x1

angela28chen commented on benchmarks/graphics_pipeline/GraphicsBenchmarkApp.cpp:
```diff
@@ -25,13 +25,23 @@ void GraphicsBenchmarkApp::InitKnobs()
     pKnobPs->SetFlagDescription("Select the pixel shader for the graphics pipeline.");
+    pKnobPs->SetIndent(1);
 
     pAllTexturesTo1x1 = GetKnobManager().CreateKnob<ppx::KnobCheckbox>("all-textures-to-1x1", false);
```
Should be indent of 2 (sorry, accidentally deleted this comment earlier)

elviscapiaq replied:
Done.

elviscapiaq commented:
I rebased the PR with the `open_xr` support for the Benchmark app.

elviscapiaq commented:
Rebased it and ready to be merged.
